e2e test failure: multiple failures of TestPeriodicDispatch_Basic

An example of the test error message:
```
Failed
=== RUN   TestE2E/Periodic/*periodic.PeriodicTest/TestPeriodicDispatch_Basic
    periodic.go:58: 
        	Error Trace:	periodic.go:58
        	Error:      	Received unexpected error:
        	            	could not inspect job: exit status 1
        	            	Error formatting the data: template: format:1:7: executing "format" at <index . 1>: error calling index: can't index item of type api.Job
        	Test:       	TestE2E/Periodic/*periodic.PeriodicTest/TestPeriodicDispatch_Basic
            --- FAIL: TestE2E/Periodic/*periodic.PeriodicTest/TestPeriodicDispatch_Basic (1.47s)
```

--- a/e2e/periodic/input/simple.nomad
+++ b/e2e/periodic/input/simple.nomad
@@ -25,4 +25,3 @@ job "periodic" {
     }
   }
 }
-

--- a/e2e/periodic/periodic.go
+++ b/e2e/periodic/periodic.go
@@ -48,15 +48,24 @@ func (tc *PeriodicTest) TestPeriodicDispatch_Basic(f *framework.F) {
 	tc.jobIDs = append(tc.jobIDs, jobID)
 
 	// register job
-	e2eutil.Register(jobID, "periodic/input/simple.nomad")
+	require.NoError(t, e2eutil.Register(jobID, "periodic/input/simple.nomad"))
 
 	// force dispatch
 	require.NoError(t, e2eutil.PeriodicForce(jobID))
 
 	// Get the child job ID
-	childID, err := e2eutil.JobInspectTemplate(jobID, `{{with index . 1}}{{printf "%s" .ID}}{{end}}`)
-	require.NoError(t, err)
-	require.NotEmpty(t, childID)
+	testutil.WaitForResult(func() (bool, error) {
+		childID, err := e2eutil.JobInspectTemplate(jobID, `{{with index . 1}}{{printf "%s" .ID}}{{end}}`)
+		if err != nil {
+			return false, err
+		}
+		if childID != "" {
+			return true, nil
+		}
+		return false, fmt.Errorf("expected non-empty periodic child jobID for job %s", jobID)
+	}, func(err error) {
+		require.NoError(t, err)
+	})
 
 	testutil.WaitForResult(func() (bool, error) {
 		status, err := e2eutil.JobInspectTemplate(jobID, `{{with index . 1}}{{printf "%s" .Status}}{{end}}`)

